### PR TITLE
Fixed #1099 - Docs in SQLite dbs never get pruned until entire db is …

### LINKED ISSUE #1099: Docs in SQLite dbs never get pruned until entire db is compacted
See https://github.com/couchbase/couchbase-lite-ios/issues/1165

### PR DESCRIPTION
**Pull request by hideki**
…compacted

- Ported from iOS: https://github.com/couchbase/couchbase-lite-ios/pull/1166

### CODE REVIEW

File: src/main/java/com/couchbase/lite/store/SQLiteStore.java [hunk 1]
```diff
@@ -289,6 +289,9 @@ else if (e.getCode() == SQLException.SQLITE_ENCRYPTION_NOTAVAILABLE)
             }
 
             if (isNew)
+                setInfo("pruned", "true"); // See -compact: for explanation
+
+            if (isNew)
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/store/SQLiteStore.java]: Should this be !isNew?

Comment (hideki) [inline src/main/java/com/couchbase/lite/store/SQLiteStore.java]: https://github.com/couchbase/couchbase-lite-ios/blob/c32147bd6e83c4c4b24b9159ff1d52ca9d0cbaf0/Source/CBL_SQLiteStorage.m#L464-L465
```objective-c
    if (isNew)
        [self setInfo: @"true" forKey: @"pruned"];  // See -compact: for explanation
```
isNew is fine.

Comment (pasin) [inline src/main/java/com/couchbase/lite/store/SQLiteStore.java]: I mean this line:
https://github.com/couchbase/couchbase-lite-ios/blob/c32147bd6e83c4c4b24b9159ff1d52ca9d0cbaf0/Source/CBL_SQLiteStorage.m#L467-L468

Comment (hideki) [inline src/main/java/com/couchbase/lite/store/SQLiteStore.java]: Thank you very much for finding this mistake. `optimizeSQLIndexes()` was never executed before.....

**Comment by hideki** [comment]
@pasin, Please re-review the code. Thanks!

**Pull request merged**